js/5etools-main.js

```diff
@@ -1331,6 +1332,261 @@ const betteR205etoolsMain = function () {
+				// Handle the language options, let user choose if needed
+				// Handles most edge cases I think
+				ret = []
+				const profEntries = Object.entries(profs);
+				for (const entry of profEntries) {
+					// Loop must be in this form -- Thanks for figuring this out Giddy
+					const [key, value] = entry;
+					if (key === "choose") {
+						// If choice is needed, call popup function
+						numChoice = 1;
+						if (value.count) numChoice = value.count;
+						const choice = await chooseProfs(value.from, numChoice, profType);
+						choice.forEach(c => ret.push(c));
+					}
+					else if (key === "anyStandard") {
+						// If any language is available, add any
+						for (i = 0; i < value; i++) ret.push("any");
+					}
+					else if (value) {
+						// If no choice is needed, add the proficiency normally
+						ret.push(key);
+					}
+				}
+				return ret;
+			}
+
+			// Get data for language proficiencies specifically
+			let backgroundLanguages = [];
+			if (bg.languageProficiencies && bg.languageProficiencies.length) {
+				if (bg.languageProficiencies.length > 1) {
+					// See Clan Crafter for an example
+					profIndex = await chooseProfsGroup(bg.languageProficiencies, "Languages");
+					backgroundLanguages = await handleProfs(bg.languageProficiencies[profIndex], "Languages");
+				}
+				else if (bg.languageProficiencies.length > 0) {
+					// Most common case
+					backgroundLanguages = await handleProfs(bg.languageProficiencies[0], "Languages");
+				}
+			}
+
+			// Tool Proficiencies
+			let backgroundTools = [];
+			if (bg.toolProficiencies && bg.toolProficiencies.length) {
+				if (bg.toolProficiencies.length > 1) {
+					// If there are different types of options
+					profIndex = await chooseProfsGroup(bg.toolProficiencies, "Tools");
+					backgroundTools = await handleProfs(bg.toolProficiencies[profIndex], "Tools")
+				}
+				else if (bg.toolProficiencies.length > 0) {
+					// Most common case
+					backgroundTools = await handleProfs(bg.toolProficiencies[0] , "Tools");
+				}
+			}
+
+			// Import items
+			async function parseItems(itemlist) {
+				const allitemList = await Renderer.item.pBuildList();
+				const x = Object.values(itemlist).map(function (item) {
+					// Returns a standardized object from a very unstandardized object
+					// Get the important variables
+					iname = "";
+					if (typeof item !== 'object') {
+						iname = item;
+					}
+					else if ('item' in item) {
+						iname = item.item;
+					}
+					else if ('special' in item) {
+						iname = item.special;
+					}
+
+					// Make the input object
+					const pareseditem = {"name": iname.split("|")[0].toTitleCase()};
+					const it = allitemList.find(pareseditem) || pareseditem;
+					// Create item data in the format importItem likes
+					const itemdata = JSON.parse(d20plus.items._getHandoutData(it)[1]);
+					// Call the importItem function usually used to import items
+					return itemdata
+				});
+
+				const y = x.map(it => {
```

Comment:
awfully clunky way of writing `x.map(it => ({subItem: JSON.stringify(it), count: 1}))`

Reply:
also these variable names all suck. `x`? `y`? Are we dealing with 2d coordinates here..?